iidm/iidm-api/src/main/java/com/powsybl/iidm/network/BoundaryPointType.java

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2024, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network;
+
+/**
+ * @author Marine Guibert {@literal <marine.guibert at artelys.com>}
+ * @author Valentin Mouradian {@literal <valentin.mouradian at artelys.com>}
+ */
+public enum BoundaryPointType {
+    HVDC,
+    PAIRED_AC,
+    UNPAIRED_AC,
+    IGNORED_AC
+}
```

Comment:
The whole "boundaryPoint" model was a WIP and has since been modified. 
We now use a simpler structure: a BoundaryTerminal record, with just a boolean to indicate if the terminal if on an AC or DC line. The paired/unpaired info can be computed when needed according to the pairing state. The IGNORED_AC value was meant to flag boundaries that we want to exclude from a computation (when computing the Net Interchange value of the Area eg). But in the end, such ignored elements just won't be added to the Area-s.